src/every_query/train.py

```
@@ -82,37 +82,48 @@ def get_step(fp: Path) -> int:
 
 
 def collate_tasks(cfg: DictConfig) -> str:
-    read_dir = f"{cfg.query.task_dir}/all"
+    task_dir = cfg.query.task_dir
+    durations = list(range(cfg.query.duration_min, cfg.query.duration_max))
 
-    task_str = f"{'|'.join(sorted(cfg.query.codes))}"
+    task_str = f"{'|'.join(sorted(cfg.query.codes))}_{'|'.join(str(d) for d in sorted(durations))}"
     hash_hex = hashlib.md5(task_str.encode()).hexdigest()
```

Ruff (0.15.4)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)